### PR TITLE
Randomly jitter quit debouncing between min, max

### LINKED ISSUE #274: QuitDebouncer needs to stagger when it eventually does quit
Else on a long netsplit, we will hammer synapse to death.  Have a min/max wait time and stagger between them, e.g. 45-60 mins.

### PR DESCRIPTION
**Pull request by lukebarnard1**
Add config to allow specification of jittering window.
When a QUIT occurs, jitter the timeout between a min and max time before bridging the leave.

Aims to fix #274

### CODE REVIEW

**Comment by leonerd** [comment]
The behaviour looks generally fine but I worry about the name of it. "jitter" usually refers to a measure of inevitable variable delay in a communication channel; i.e. something that just happens. Whereas here we're causing it. Can you rename it something like "stagger" or "delay"?

File: config.sample.yaml [hunk 1]
```diff
@@ -58,11 +58,17 @@ ircService:
...
+        # a netsplit. Debouncing is jittered randomly between startMs and endMs so that the HS
+        # is not sent many requests to leave rooms all at once if a netsplit occurs and many
+        # people to not rejoin.
         # If the user with the same IRC nick as the one who sent the quit rejoins a channel
-        # before delayMs, consider them back online and do not bridge the quit.
+        # they are considered back online and the quit is not bridged, so long as the rejoin
+        # occurs before the randomly-jittered timeout is not reached.
         # Default: 3600000, = 1h
-        delayMs: 3600000 # 1h
+        startMs: 3600000 # 1h
```
Comment (leonerd) [inline config.sample.yaml]: Maybe `delayMinMs` and `delayMaxMs` ?

**Comment by lukebarnard1** [comment]
I think "jitter" describes it perfectly and is something that can be done on purpose.

**Comment by leonerd** [comment]
LGTM

**Pull request merged**